src/roadstop.cpp

```diff
@@ -277,9 +277,16 @@ bool RoadStop::Enter(RoadVehicle *rv)
  * Leave the road stop
  * @param rv the vehicle that leaves the stop
  */
-void RoadStop::Entry::Leave(const RoadVehicle *rv)
+void RoadStop::Entry::Leave()
 {
-	this->occupied -= rv->gcache.cached_total_length;
+	this->vehicles -= 1;
+	this->vehicles = std::max(this->vehicles, 0);
+
+	if(this->vehicles == 0){
```

Comment:
Missing spaces, although this statement could be inline.
```suggestion
	if (this->vehicles == 0) {
```

Or:
```suggestion
	if (this->vehicles == 0) this->occupied = 0;
```

It would also be nice to have a comment explaining why we don't clear occupancy until the stop is empty, because future us looking at the code won't know to find this PR. 😃